### PR TITLE
Disable optimizations on CI builds.

### PR DESCRIPTION
**Pull request by patrickt**
Build times are killing us. This gets us down to 9 minutes from 40. Docker builds should be unaffected because we don’t pass --disable-optimizations to new-configure in the Dockerfile.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,12 +28,12 @@ before_install:
 - cabal --version
 
 install:
-- cabal new-update -v hackage.haskell.org
-- cabal new-configure --enable-tests --write-ghc-environment-files=always
-- cabal new-build --only-dependencies -j
+- cabal new-update -v
+- cabal new-configure --enable-tests --disable-optimization --write-ghc-environment-files=always --jobs=2
```
Comment (robrix) [inline .travis.yml]: Why `--jobs=2` instead of `-j`?

File: .travis.yml [hunk 2]
```diff
@@ -28,12 +28,12 @@ before_install:
...
+- cabal new-build --only-dependencies
 
 script:
-- cabal new-build -j
+- cabal new-build
```
Comment (robrix) [inline .travis.yml]: Moving that to the `new-configure` step is a great idea. We should do this in `fused-effects` too.

File: .travis.yml (hunk 1 above)
Comment (patrickt) [reply .travis.yml]: `-j` means `--jobs=NCPUS`. I was trying to see if 2 made things go any faster, since Travis images have one core visible.

Comment (robrix) [reply .travis.yml]: 👍 thanks!

**Pull request merged**